src/extensions/scratch3_gdx_for/index.js

```diff
@@ -71,49 +92,79 @@ class GdxFor {
      * Called by the runtime when user wants to scan for a peripheral.
      */
     scan () {
-        if (this._ble) {
-            this._ble.close();
+        if (this._device) {
+            this._device.close();
         }
 
-        navigator.bluetooth.requestDevice({
-            filters: [{namePrefix: 'GDX'}],
-            optionalServices: ['d91714ef-28b9-4f91-ba16-f0d9a604f112']
-        }).then(bleWbtDevice => createDevice(bleWbtDevice, {open: false}))
-            .then(device => {
-                this._ble = device;
-                this._ble.open(false); // false prevents starting of measurements
-                this._ble.on('device-opened', this._onConnect);
-                this._ble.on('device-closed', this._onClose);
-            });
+        this._scratchLinkSocket = new BLE(this._runtime, this._extensionId, {
+            filters: [
+                {namePrefix: 'GDX-FOR'}
+            ],
+            optionalServices: [
+                BLEUUID.service
+            ]
+        }, this._onConnect);
     }
 
     /**
      * Called by the runtime when user wants to connect to a certain peripheral.
      * @param {number} id - the id of the peripheral to connect to.
      */
     connect (id) {
```

Comment:
Is this function also defined on line #145?

Reply:
it is XD resolved